### PR TITLE
W-9172445 Adding pre_processing_step

### PR DESCRIPTION
**Pull request by Ullimague**
@W-9172445@
@rev jakemannix@
@rev balikasg@
@rev lastmansleeping@

### CODE REVIEW

**Comment by lastmansleeping** [comment]
> @W-9172445@

@Ullimague Good to not have internal artifacts on ml4ir I think.

**Review by lastmansleeping — COMMENTED** [review]
@Ullimague You might want to do the following while you are it as well:
- bump ml4ir version in setup.py
- Update changelog -> https://github.com/salesforce/ml4ir/blob/master/docs/source/misc/changelog.md

**Review by lastmansleeping — APPROVED** [review]
LGTM

File: python/ml4ir/base/pipeline.py [hunk 1]
```diff
@@ -439,6 +436,13 @@ def run(self):
 
         return experiment_tracking_dict
 
+    def pre_processing_step(self):
+        """
+        Performs any pre processing step (such as copying data) by custom class inheriting this class, which would
```
Comment (balikasg) [inline python/ml4ir/base/pipeline.py:441]: this is a somewhat cryptic docstring. I would go with something like: "Performs arbitrary pre-processing steps such as copying or transforming data that the rest of the code can not accommodate. It serves as a placeholder without an explicit implementation (returns `self`) in the `base` pipeline. We expect that users can extend it in their custom pipelines."

**Review by balikasg — APPROVED** [review]
LGTM. I would suggest to take 2' to improve a bit the docstring, but it is not a blocking comment on your implementation.

**Pull request merged**